dev/tracing_tests/test/image_painting_event_test.dart

```diff
@@ -0,0 +1,92 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:convert' show jsonEncode;
+import 'dart:developer' as developer;
+import 'dart:typed_data';
+import 'dart:ui' as ui;
+
+import 'package:flutter/painting.dart';
+import 'package:flutter/widgets.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:vm_service/vm_service.dart';
+import 'package:vm_service/vm_service_io.dart';
+
+void main() {
+  VmService vmService;
+  LiveTestWidgetsFlutterBinding binding;
+  setUpAll(() async {
+    final developer.ServiceProtocolInfo info =
+        await developer.Service.getInfo();
+
+    if (info.serverUri == null) {
+      fail('This test _must_ be run with --enable-vmservice.');
+    }
+
+    vmService = await vmServiceConnectUri('ws://localhost:${info.serverUri.port}${info.serverUri.path}ws');
+    await vmService.streamListen(EventStreams.kExtension);
+
+    // Initialize bindings
+    binding = LiveTestWidgetsFlutterBinding();
+    binding.framePolicy = LiveTestWidgetsFlutterBindingFramePolicy.fullyLive;
+    binding.attachRootWidget(const SizedBox.expand());
+  });
+
+  test('Image painting events', () async {
+    String extensionKind;
+    String extensionData;
+    vmService.onExtensionEvent.listen((Event event) {
+      extensionKind = event.extensionKind;
+      extensionData = jsonEncode(event.extensionData.data);
+    });
+
+    const TestImage image = TestImage(width: 300, height: 300);
+    final TestCanvas canvas = TestCanvas();
+    paintImage(
+      canvas: canvas,
+      rect: const Rect.fromLTWH(50.0, 75.0, 200.0, 100.0),
+      image: image,
+      debugImageLabel: 'test.png',
+    );
+
+    // Make sure that we don't report an identical image size info if we
+    // redraw in the next frame.
+    // tester.binding.scheduleForcedFrame();
+    // await tester.pump();
+    await binding.endOfFrame;
+
+    paintImage(
+      canvas: canvas,
+      rect: const Rect.fromLTWH(50.0, 75.0, 200.0, 100.0),
+      image: image,
+      debugImageLabel: 'test.png',
+    );
+    await binding.endOfFrame;
```

Comment:
paint a couple identical images as part of this test to verify whether output dedupes or not.

Reply:
That's what's happening here - as long as they're in adjacent frames, they're getting deduplicated. It's the same image painted twice here.

Are you saying you'd like to see deduplicating in a single frame?

Reply:
yeah I was interested in a duplicate within a single frame as well as on the devtools size we don't care if the same image was rendered multiple times and we only care about knowing the largest size it was rendered at. there might be some edge cases where you have one image rendered in a bunch of places in a frame.

Reply:
Ahh ok - I've reworked it so it will dedupe to the largest image in a frame, and added a test.